newIDE/app/src/AssetStore/CreditsPackages/CreditsPackageStoreContext.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import * as React from 'react';
+import {
+  listListedCreditsPackages,
+  type CreditsPackageListingData,
+} from '../../Utils/GDevelopServices/Shop';
+
+type CreditsPackageStoreState = {|
+  fetchCreditsPackages: () => void,
+  creditsPackageListingDatas: ?Array<CreditsPackageListingData>,
+  error: ?Error,
+|};
+
+export const CreditsPackageStoreContext = React.createContext<CreditsPackageStoreState>(
+  {
+    fetchCreditsPackages: () => {},
+    creditsPackageListingDatas: null,
+    error: null,
+  }
+);
+
+type CreditsPackageStoreStateProviderProps = {|
+  children: React.Node,
+|};
+
+export const CreditsPackageStoreStateProvider = ({
+  children,
+}: CreditsPackageStoreStateProviderProps) => {
+  const [error, setError] = React.useState<?Error>(null);
+  const [
+    creditsPackageListingDatas,
+    setCreditsPackageListingDatas,
+  ] = React.useState<?Array<CreditsPackageListingData>>(null);
+
+  const isLoading = React.useRef<boolean>(false);
+
+  const fetchCreditsPackages = React.useCallback(
+    () => {
+      // If the credit packages are already loaded, don't load them again.
+      if (isLoading.current || creditsPackageListingDatas) return;
+
+      (async () => {
+        setError(null);
+        isLoading.current = true;
+
+        try {
+          const fetchedCreditsPackageListingDatas = await listListedCreditsPackages();
+
+          console.info(
+            `Loaded ${
+              fetchedCreditsPackageListingDatas
+                ? fetchedCreditsPackageListingDatas.length
+                : 0
+            } credit packages from the store.`
+          );
+
+          setCreditsPackageListingDatas(fetchedCreditsPackageListingDatas);
+        } catch (error) {
+          console.error(
+            `Unable to load the credit packages from the store:`,
+            error
+          );
+          setError(error);
+        }
+
+        isLoading.current = false;
+      })();
+    },
+    [isLoading, creditsPackageListingDatas]
+  );
+
+  React.useEffect(
+    () => {
+      if (isLoading.current) return;
+
+      const timeoutId = setTimeout(() => {
+        console.info('Pre-fetching credit packages from the store...');
+        fetchCreditsPackages();
+      }, 5000);
+      return () => clearTimeout(timeoutId);
+    },
+    [fetchCreditsPackages, isLoading]
```

Comment:
You shouldn't need to put isLoading in the dependencies. And if you do, you should use a state variable I think

Reply:
This logic is copied from other stores (assets, templates...)
I'll remove the isLoading as indeed it looks useless.
I've double checked what you're mentioning but only see 1 call happening so we're probably fine.